src/containers/BooksList.js

```diff
@@ -3,45 +3,60 @@ import '../styles/BooksList.scss';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import Book from '../components/Book';
-import { removeBook } from '../actions/index';
+import CategoryFilter from '../components/CategoryFilter';
+import { removeBook, filterBook } from '../actions/index';
+import filteredBooks from '../logic/filter';
 
-const BooksList = ({ books, removeBook }) => (
-  <table>
-    <thead>
-      <tr>
-        <th>Book ID</th>
-        <th>Title</th>
-        <th>Category</th>
-        <th>Remove</th>
-      </tr>
-    </thead>
-    <tbody>
-      {
-          books.map(book => (
-            <Book
-              key={book.id}
-              title={book.title}
-              category={book.category}
-              id={book.id}
-              removeBook={removeBook}
-            />
-          ))
-        }
-    </tbody>
-  </table>
+const BooksList = ({
+  books, filter, removeBook, filterBook,
+}) => (
+  <div>
+    <CategoryFilter handleFilterChange={filterBook} />
+    <table>
+      <thead>
+        <tr>
+          <th>Book ID</th>
+          <th>Title</th>
+          <th>Category</th>
+          <th>Remove</th>
+        </tr>
+      </thead>
+      <tbody>
+        {
+            filteredBooks(filter, books).map(book => (
+              <Book
+                key={book.id}
+                title={book.title}
+                category={book.category}
+                id={book.id}
+                removeBook={removeBook}
+              />
+            ))
+          }
```

Comment:
**OPTIONAL SUGGESTION**
Here you can make use of another React hook, [useMemo](https://pt-br.reactjs.org/docs/hooks-reference.html#usememo).